db.go

```diff
@@ -319,3 +319,14 @@ func isSlice(possibleSlice interface{}) bool {
 
 	return slice
 }
+
+// IsDuplicateUniqueIdx returns true if the error was caused by
+// trying to create a record with a field that is duplicated
+// in an unique index.
+func IsDuplicateUniqueIdx(err error) bool {
+	if err == nil {
+		return false
+	}
+	match, _ := regexp.MatchString(`Database index .* already contains .*, with record .*`, err.Error())
```

Comment:
ToDo:
Create ticket on SurrealDB project to include error code in error response.